UpdatedSyntheticDataset/SyntheticDataset2/ImageCreator/modular_target.py

```diff
@@ -0,0 +1,92 @@
+from PIL import Image
+import math, random, json
+from .settings import Settings
+from .modular_target_settings import ModularTargetSettings
+from .specified_target import SpecifiedTarget
+from SyntheticDataset2.ElementsCreator.shape_types import ShapeTypes
+from SyntheticDataset2.ElementsCreator.shape_orientations import ShapeOrientations
+
+class ModularTarget(object):
+
+    def __init__(self):
+        self.size = Settings.TARGET_GENERATION_SIZE_IN_PIXELS
+        self.proportionality_range = Settings.PROPORTIONALITY_RANGE
+        self.pixelization_level = Settings.PIXELIZATION_LEVEL
+        self.noise_level = Settings.NOISE_LEVEL
+
+        if (ModularTargetSettings.RANDOM_SHAPE_TYPE):
+            shape_type_list = [ShapeTypes.CIRCLE, ShapeTypes.QUARTERCIRCLE, ShapeTypes.HALFCIRCLE,
+                               ShapeTypes.CROSS, ShapeTypes.TRIANGLE, ShapeTypes.SQUARE,
+                               ShapeTypes.RECTANGLE, ShapeTypes.TRAPEZOID, ShapeTypes.PENTAGON,
+                               ShapeTypes.HEXAGON, ShapeTypes.HEPTAGON, ShapeTypes.OCTAGON,
+                               ShapeTypes.STAR]
+
+            self.shape_type = shape_type_list[random.randint(0, 12)]
```

Comment:
Replace the random with the following: ```random.randint(len(shape_type_list))```. This way, if we change the potential shape types, we do not have to recode this random assignment.